compiler-core/src/language_server/tests/action.rs

```diff
@@ -751,6 +763,69 @@ fn test_convert_outer_let_assert_to_case() {
     ));
 }
 
+#[test]
+fn move_imports_to_the_top_of_the_module_does_not_pop_up_if_no_imports_can_be_moved() {
+    let code = r#"import error
+pub fn main() {
+  error.is_ok()
+}"#;
+
+    assert!(engine_response(code, 0)
+        .result
+        .expect("ok response")
+        .is_none());
+}
+
+#[test]
+fn move_imports_to_the_top_of_the_module_does_not_pop_up_if_not_over_an_import() {
+    let code = r#"pub fn main() {}
+
+const wibble = 1
+
+import list
+"#;
+
+    assert!(engine_response(code, 0)
+        .result
+        .expect("ok response")
+        .is_none());
+
+    assert!(engine_response(code, 1)
+        .result
+        .expect("ok response")
+        .is_none());
+
+    assert!(engine_response(code, 2)
+        .result
+        .expect("ok response")
+        .is_none());
+}
+
+#[test]
+fn move_imports_to_the_top_of_the_module_moves_all_imports_when_on_import_below_a_definition_1() {
+    let code = r#"import list
+
+pub fn main() {}
+
+import result
+import map
+"#;
+
+    insta::assert_snapshot!(apply_first_code_action_with_title(code, 4, MOVE_IMPORTS_UP));
+}
+
+#[test]
+fn move_imports_to_the_top_of_the_module_moves_all_imports_when_on_import_below_a_definition_2() {
+    let code = r#"import list
+
+pub fn main() {}
+
+import result
```

Comment:
What happens if the imports are lower but in groups? Do the groups get preserved?

What about comments?

Reply:
Mmh this is quite tricky, basically I'd have to reimplement part of the pretty printer here but I'm not a fan of that... I'll have to think a bit more how to approach this